docs/contributing/code-contributions.md

```diff
@@ -2,110 +2,102 @@
+  the built files from your cloned copy of Gatsby. It'll watch for your changes
+  to Gatsby packages and copy them into the site. For more detailed instructions
+  see the [gatsby-dev-cli README](https://www.npmjs.com/package/gatsby-dev-cli) and check out the [gatsby-dev-cli demo video](https://www.youtube.com/watch?v=D0SwX1MSuas).
+
+  - Note: If you plan to modify packages that are exported from `gatsby` directly, you need to either add those manually to your test sites so that they are listed in `package.json` (e.g. `yarn add gatsby-link`), or specify them explicitly with `gatsby-dev --packages gatsby-link`).
 
-To contribute example sites, it is recommended to create your own GitHub repo and link to it from your source plugin, etc.
-
-## Using Docker to set up test environments
-
-With all of the possible Gatsby integrations, it might help to spin up a Docker container with the software application you need to test. This makes installation a breeze, so you can focus less on getting set up and more on the integration details that matter to you.
-
-> Do you have a setup not listed here? Let us know by adding it to this file and opening a PR.
-
-### Docker, WordPress and Gatsby
-
-To install WordPress to use with Gatsby, this `docker-compose.yml` file will come in handy:
-
-```yaml:title=docker-compose.yml
-version: "2"
-
-services:
-  db:
-    image: mysql:5.6
-    container_name: sessions_db
-    ports:
-      - "3306:3306"
-    volumes:
-      - "./.data/db:/var/lib/mysql"
-    restart: always
-    environment:
-      MYSQL_ROOT_PASSWORD: wordpress
-      MYSQL_DATABASE: wordpress
-      MYSQL_USER: wordpress
-      MYSQL_PASSWORD: wordpress
-
-  wordpress:
-    image: wordpress:latest
-    container_name: sessions_wordpress
-    depends_on:
-      - db
-    links:
-      - db
-    ports:
-      - "7000:80"
-    restart: always
-    environment:
-      WORDPRESS_DB_HOST: db:3306
-      WORDPRESS_DB_PASSWORD: wordpress
-    volumes:
-      - ./wp-content:/var/www/html/wp-content
-      - ./wp-app:/var/www/html
-
-  phpmyadmin:
-    image: phpmyadmin/phpmyadmin
-    container_name: sessions_phpmyadmin
-    environment:
-      - PMA_ARBITRARY=1
-      - PMA_HOST=sessions_db
-      - PMA_USER=wordpress
-      - PMA_PASSWORD=wordpress
-    restart: always
-    ports:
-      - 8080:80
-    volumes:
-      - /sessions
+- If you've recently run `gatsby-dev` your `node_modules` will be out of sync with current published packages. In order to undo this, you can remove the `node_modules` directory or run:
+
+```shell
+git checkout package.json; yarn --force
 ```
 
-Use the above file contents when following the Docker WordPress install instructions: https://docs.docker.com/compose/wordpress/
+### Add tests
 
-Using Docker Compose, you can start and stop a WordPress instance and integrate it with the [Gatsby WordPress source plugin](/docs/sourcing-from-wordpress/).
+- Add tests and code for your changes.
```

Comment:
It could be useful expand this section (or create a new page) to answer questions like:

- What tests should I add and when (e.g. unit, integration, e2e)?
- Where do the different types of tests live?
- What are some common issues I could run into trying to run tests and how do I fix them?